--- locallib.php	2012-09-24 09:35:54.000000000 +0500
+++ locallib.php	2012-09-24 09:41:09.031022887 +0500
@@ -22,7 +22,7 @@ function get_lecture_sessions( $course,$
 {
         global $CFG, $DB;
         $endtime =strtotime(date("Y-m-d", $endtime) . " +1 days");
-        $sql="select count(*) from {attendance_sessions} where courseid=$course->id and lasttaken IS NOT NULL and description IN(1,4) and sessdate >=$course->startdate and sessdate<=$endtime";
+        $sql="select count(*) from {attendance_sessions} where courseid=$course->id and lasttaken IS NOT NULL and description IN(1,4,5,6) and sessdate >=$course->startdate and sessdate<=$endtime";
 	$sessions=$DB->count_records_sql($sql,array());
         return $sessions;
 
@@ -111,7 +111,7 @@ function get_attendance($userid, $course
 
 function get_attendance_bydate($userid, $course, $statusid=0, $where='')
 {
-/*	global $CFG;
+	/*global $CFG;
 	global $DB;
 	$qry = "SELECT count(*) as cnt
 		  	  FROM {$CFG->prefix}attendance_log al
@@ -128,17 +128,18 @@ function get_attendance_bydate($userid,
         
 
 	return $DB->count_records_sql($qry);*/
-global $CFG;
+
+	global $CFG;
 	global $DB;
 	$qry = "SELECT l.id, l.statusid, l.statusset, s.description
 		  	  FROM {$CFG->prefix}attendance_log l
 			  JOIN {$CFG->prefix}attendance_sessions s
 			    ON l.sessionid = s.id
 			 WHERE ".$where." ";
-                                //ats.courseid = $course->id
-			  	//AND ats.sessdate >= $course->startdate
+	//ats.courseid = $course->id
+	//AND ats.sessdate >= $course->startdate
 
-	         	$qry .= " AND l.studentid = $userid";
+	$qry .= " AND l.studentid = $userid";
 	if ($statusid) {
 		$qry .= " AND l.statusid = $statusid";
 	}
@@ -149,36 +150,43 @@ global $CFG;
 		$stat_grades = $DB->get_records_menu('attendance_statuses', array('courseid'=>$course->id), 'id', 'id, grade');
 		foreach ($logs as $log) {
 			$ids = array_flip(explode(',', $log->statusset));
-//			$grades = array_intersect_key($stat_grades, $ids); // require PHP 5.1.0 and higher
+			//			$grades = array_intersect_key($stat_grades, $ids); // require PHP 5.1.0 and higher
 			$grades = local_array_intersect_key($stat_grades, $ids); //temporary solution, for support PHP 4.3.0 which minimal requirement for Moodle 1.9.x
-                        if($log->description=="1")
-                            $maxgrade += max($grades);
-                        if($log->description=="2")
-                            $maxgrade += max($grades);
-                        if($log->description=="3")
-                            $maxgrade += max($grades);
-                        if($log->description=="4")
-                            $maxgrade += 1.5;
-			
+			if($log->description=="1")
+			$maxgrade += max($grades);
+			if($log->description=="2")
+			$maxgrade += max($grades);
+			if($log->description=="3")
+			$maxgrade += max($grades);
+			if($log->description=="4")
+			$maxgrade += 1.5;
+			if($log->description=="5")
+			$maxgrade +=2;
+			if($log->description=="6")
+			$maxgrade += 3;
+
 		}
 		if(isset($course->attendance_margin)){
 			$margin = $course->attendance_margin;
 			while($margin > 0){
 				if($log->description=="1")
 				$maxgrade += 1;
-	            if($log->description=="2")
-					$maxgrade += 1;
-	            if($log->description=="3")
-					$maxgrade += 1;
+				if($log->description=="2")
+				$maxgrade += 1;
+				if($log->description=="3")
+				$maxgrade += 1;
 				if($log->description == "4")
-					$maxgrade += 1.5;
+				$maxgrade += 1.5;
+				if($log->description  =="5")
+				$maxgrade +=2;
+				if($log->description=="6")
+				$maxgrade += 3;
 				$margin--;
 			}
 		}
 	}
-	
-	return $maxgrade;
 
+	return $maxgrade;
 }
 
 function get_firstmonday($month,$year){
@@ -215,6 +223,10 @@ function get_grade($userid, $course)
 				$result += $stat_grades[$log->statusid];
 			if($log->description == "4")
 				$result += $stat_grades[$log->statusid]*1.5;
+	if($log->description=="5")
+			$result += $stat_grades[$log->statusid]*2;
+			if($log->description == "6")
+			$result += $stat_grades[$log->statusid]*3;
 		}
 		
 		if(isset($course->attendance_margin)){
@@ -228,6 +240,10 @@ function get_grade($userid, $course)
 					$result += 1;
 				if($log->description == "4")
 					$result += 1.5;
+		if($log->description=="5")
+				$result += 2;
+				if($log->description == "6")
+				$result += 3;
 				$margin--;
 			}
 		}
@@ -255,8 +271,13 @@ function get_grade_bydate($userid, $cour
 				$result += $stat_grades[$log->statusid];
             if($log->description=="3")
 				$result += $stat_grades[$log->statusid];
-			if($log->description=="4")
+	    if($log->description=="4")
 				$result += $stat_grades[$log->statusid]*1.5;
+	if($log->description=="5")
+			$result += $stat_grades[$log->statusid]*2;
+			if($log->description=="6")
+			$result += $stat_grades[$log->statusid]*3;
+	
 		}
 	}
 
@@ -299,6 +320,10 @@ function get_maxgrade($userid, $course)
                             $maxgrade += max($grades);
                         if($log->description=="4")
                             $maxgrade += 1.5;
+			if($log->description=="5")
+			$maxgrade += 2;
+			if($log->description=="6")
+			$maxgrade += 3;
 			
 		}
 		if(isset($course->attendance_margin)){
@@ -312,6 +337,10 @@ function get_maxgrade($userid, $course)
 					$maxgrade += 1;
 				if($log->description == "4")
 					$maxgrade += 1.5;
+		if($log->description=="5")
+				$maxgrade += 2;
+				if($log->description == "6")
+				$maxgrade += 3;
 				$margin--;
 			}
 		}
@@ -347,6 +376,10 @@ function get_maxgrade_bydate($userid, $c
 				$maxgrade += max($grades);
 			if($log->description=="4")
 				$maxgrade += 1.5;
+			if($log->description=="5")
+			$maxgrade += 2;
+			if($log->description=="6")
+			$maxgrade += 3;
 		}
 	}
         //echo $qry."<br>";
@@ -597,6 +630,12 @@ function get_description($description, $
     		case "4":
     			return "90M-Class";
     			break;
+		case "5":
+				return "2H-Class";
+				break;
+			case "6":
+				return "3H-Class";
+				break;
     	}
     }else{
         if($description == "1"){
@@ -607,7 +646,11 @@ function get_description($description, $
             return get_string('threehourslab1','attforblock');
         }elseif($description == "4"){
             return get_string('ninety_minsclass1','attforblock');
-        }else{
+        }elseif($description == "5"){
+			return get_string('twohoursclass','attforblock');
+		}elseif($description == "6"){
+			return get_string('threehoursclass','attforblock');
+		}else{
             return $description;
         }
     }
